lib/Dialect/Torch/Transforms/DecomposeComplexOpsEarly.cpp

```diff
@@ -0,0 +1,279 @@
+#include "torch-mlir/Dialect/Torch/Transforms/Passes.h"
+#include "torch-mlir/Dialect/Torch/Utils/Utils.h"
+#include "llvm/ADT/SmallVector.h"
+#include "llvm/ADT/StringExtras.h"
+#include <cstdint>
+#include <numeric>
+
+using namespace mlir;
+using namespace mlir::torch;
+using namespace mlir::torch::Torch;
+
+namespace {
+class DecomposeAtenSplitSizesOp : public OpRewritePattern<AtenSplitSizesOp> {
+public:
+  using OpRewritePattern::OpRewritePattern;
+  LogicalResult matchAndRewrite(AtenSplitSizesOp op,
+                                PatternRewriter &rewriter) const override {
+    SmallVector<int64_t> splitSizes;
+    // TODO: Support non static splitSizes.
+    if (!matchPattern(op.split_size(), m_TorchConstantIntList(splitSizes)))
+      return failure();
+
+    int64_t dim;
+    if (!matchPattern(op.dim(), m_TorchConstantInt(&dim)))
+      return rewriter.notifyMatchFailure(op, "op.split_sizes() must be a const int list");
+
+    auto listType = op.getType().template cast<Torch::ListType>();
+    Location loc = op.getLoc();
+
+    // Calculate the end indices of the slices.
+    SmallVector<int64_t> endIdx;
+    endIdx.reserve(splitSizes.size());
+    std::partial_sum(splitSizes.begin(), splitSizes.end(), endIdx.begin());
+    
+    int64_t start = 0;
+    SmallVector<Value> listOfTensors;
+    Value stepVal = rewriter.create<Torch::ConstantIntOp>(
+        loc, rewriter.getI64IntegerAttr(1));
+
+    // For each of the split create slice operation to slice the input tensor
+    // from `start` to `endIdx`.
+    for (unsigned i = 0; i < splitSizes.size(); i++) {
+      Value startVal = rewriter.create<Torch::ConstantIntOp>(
+          loc, rewriter.getI64IntegerAttr(start));
+      Value endVal = rewriter.create<Torch::ConstantIntOp>(
+          loc, rewriter.getI64IntegerAttr(endIdx[i]));
+      Type resultType = listType.getContainedType();
+      Value slice = rewriter.create<AtenSliceTensorOp>(
+          loc, resultType, op.self(), op.dim(), startVal, endVal, stepVal);
+      listOfTensors.push_back(slice);
+      start = endIdx[i];
+    }
+
+    Value result = rewriter.create<PrimListConstructOp>(
+        loc, Torch::ListType::get(listType.getContainedType()), listOfTensors);
+
+    rewriter.replaceOp(op, result);
+    return success();
+  }
+};
+} // namespace
+
+namespace {
+
+template <typename SplitTensorLikeOp>
+class DecomposeAtenSplitTensorLikeOp
+    : public OpRewritePattern<SplitTensorLikeOp> {
+public:
+  using OpRewritePattern<SplitTensorLikeOp>::OpRewritePattern;
+  LogicalResult matchAndRewrite(SplitTensorLikeOp op,
+                                PatternRewriter &rewriter) const override {
+    int64_t dim;
+    if (!matchPattern(op.dim(), m_TorchConstantInt(&dim)))
+      return rewriter.notifyMatchFailure(op, "dim must be constant");
+
+    auto listType = op.getType().template cast<Torch::ListType>();
+    Location loc = op.getLoc();
+
+    auto viewOp = dyn_cast<AtenViewOp>(op.self().getDefiningOp());
```

Comment:
Currently this conversion pattern only works if the input is the result of an `AtenViewOp`. This is too strict of a requirement. You should be able to make the calculations for the size of the chunks by using `getDimOp` to get the size of the dimension being used for splitting.

Reply:
So if I'm getting this right, I would remove the auto viewOp, and would replace 
`autosizesListOp =
        dyn_cast<PrimListConstructOp>(viewOp.size().getDefiningOp());` 
with
`auto sizesListOp =
        dyn_cast<PrimListConstructOp>(op.self().getDimOp());`
Is that right?

Reply:
No, `getDimOp` is not a method of `op.self()`, it's a helper function. You can find example usage by running `git grep 'getDimOp'`. You then have to build a `PrimListConstructOp` with the dimensions rather than doing `dyn_cast<PrimListConstructOp>`

Reply:
I am wondering something about this pass. This implementation requires that the dim size for the split dimension is static, however because the DecomposeComplexOpsEarlyPass happens before shape refinement, if I understand correctly, we almost never know the shape of this dimension (except when we have something like a defining view op), even if it is something that would be determined statically. I think this was the motivation for requiring an `AtenViewOp` to define the input tensor. @ramiro050 Does this sound right to you?